### PR TITLE
Mirage solo5 0.6.0 components

### PR DESCRIPTION
**Pull request by mato**
This is an "omnibus" PR with the component packages for Mirage/Solo5 0.6.0.

Additionally, I have added some upper bounds on the existing mirage-solo5.0.6.0 to make later breaking releases easier.

Part of mirage/mirage#992.

### CODE REVIEW

File: packages/mirage-block-solo5/mirage-block-solo5.0.6.0/opam [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+build: [
+ ["dune" "subst"] {pinned}
+ ["dune" "build" "-p" name "-j" jobs]
+ ["dune" "runtest" "-p" name] {with-test}
+]
+
+depends: [
+  "ocaml" {>= "4.04.2"}
+  "dune"  {build & >= "1.0"}
```
Comment (mseri) [inline packages/mirage-block-solo5/mirage-block-solo5.0.6.0/opam]:
```suggestion
  "dune"  {>= "1.0"}
```

File: packages/mirage-bootvar-solo5/mirage-bootvar-solo5.0.6.0/opam [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  "org:mirage"
+]
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
+]
+depends: [
+  "ocaml" {>= "4.04.2"}
+  "dune" {build & >= "1.0"}
```
Comment (mseri) [inline packages/mirage-bootvar-solo5/mirage-bootvar-solo5.0.6.0/opam]:
```suggestion
  "dune" {>= "1.0"}
```

File: packages/mirage-console-solo5/mirage-console-solo5.0.6.0/opam [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  "org:mirage"
+]
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
+]
+depends: [
+  "ocaml" {>= "4.04.2"}
+  "dune" {build & >= "1.0"}
```
Comment (mseri) [inline packages/mirage-console-solo5/mirage-console-solo5.0.6.0/opam]:
```suggestion
  "dune" {>= "1.0"}
```

File: packages/mirage-block-solo5/mirage-block-solo5.0.6.0/opam (hunk 1 above)
Comment (mato) [reply packages/mirage-block-solo5/mirage-block-solo5.0.6.0/opam]: @mseri Actually, I misread this. Why would this not be a build dependency only?

**Comment by mato** [comment]
The OpenSUSE failures are likely due to missing depexts on solo5-bindings-*, but I have no idea what the SUSE depexts should be.

File: packages/mirage-block-solo5/mirage-block-solo5.0.6.0/opam (hunk 1 above)
Comment (avsm) [reply packages/mirage-block-solo5/mirage-block-solo5.0.6.0/opam]: dune is not forward compatible, so it needs to recompile all packages if you downgrade dune.  Making it `{build}` has never really worked, so we've removed them all from the main opam repository.

**Comment by avsm** [comment]
That's a strange OpenSUSE failure mode for a missing depext:

```
# mkdir -p /home/opam/.opam/4.09/lib/pkgconfig \
#     /home/opam/.opam/4.09/lib/solo5-bindings-spt \
#     /home/opam/.opam/4.09/include/solo5-bindings-spt/solo5 \
#     /home/opam/.opam/4.09/include/solo5-bindings-spt/crt
# cp -R include/solo5 include/crt /home/opam/.opam/4.09/include/solo5-bindings-spt
# cp bindings/spt/solo5_spt.o bindings/spt/solo5_spt.lds \
#     /home/opam/.opam/4.09/lib/solo5-bindings-spt
# cp: cannot stat 'bindings/spt/solo5_spt.o': No such file or directory
```

Does the build just silently skip solo5_spt.o? I'd expect a failure on trying to build the object file, not installing it.

**Comment by mato** [comment]
> That's a strange OpenSUSE failure mode for a missing depext:
> 
> ```
> # mkdir -p /home/opam/.opam/4.09/lib/pkgconfig \
> #     /home/opam/.opam/4.09/lib/solo5-bindings-spt \
> #     /home/opam/.opam/4.09/include/solo5-bindings-spt/solo5 \
> #     /home/opam/.opam/4.09/include/solo5-bindings-spt/crt
> # cp -R include/solo5 include/crt /home/opam/.opam/4.09/include/solo5-bindings-spt
> # cp bindings/spt/solo5_spt.o bindings/spt/solo5_spt.lds \
> #     /home/opam/.opam/4.09/lib/solo5-bindings-spt
> # cp: cannot stat 'bindings/spt/solo5_spt.o': No such file or directory
> ```
> 
> Does the build just silently skip solo5_spt.o? I'd expect a failure on trying to build the object file, not installing it.

Yeah, it's due to https://github.com/Solo5/solo5/issues/418. I don't really want to cut another Solo5 release just for that.

File: packages/mirage-block-solo5/mirage-block-solo5.0.6.0/opam (hunk 1 above)
Comment (mato) [reply packages/mirage-block-solo5/mirage-block-solo5.0.6.0/opam]: > dune is not forward compatible ...
Ok, done in 73f88dd1de.

**Comment by mato** [comment]
I tried adding libseccomp-devel to the depexts for SUSE, but it doesn't seem to help. Is there any way I can pull the docker image the CI is using, to figure out what is going on?

**Comment by mato** [comment]
So, unlike every other Linux distribution in our mix, OpenSUSE installs `seccomp.h` as `libseccomp/seccomp.h`. Handling these kinds of setups will require some surgery on the Solo5 build system, which I've reported as https://github.com/Solo5/solo5/issues/419. 

In the interests of getting a Mirage/Solo5 release out, can we just ignore this for now and merge please? It can be fixed later in a Solo5 point release.

**Comment by hannesm** [comment]
i agree with @mato and see a green button here :D

**Pull request merged**